nano/node/transport/tcp.cpp

```diff
@@ -514,6 +566,12 @@ void nano::transport::tcp_channels::start_tcp_receive_node_id (std::shared_ptr<n
 										{
 											callback_a (channel_a);
 										}
+										// Listen for possible responses
+										channel_a->response_server = std::make_shared<nano::bootstrap_server> (channel_a->socket, node_l);
+										channel_a->response_server->keepalive_first = false;
+										channel_a->response_server->type = nano::bootstrap_server_type::realtime_response_server;
+										channel_a->response_server->remote_node_id = channel_a->get_node_id ().get ();
```

Comment:
It seems like bootstrap_server should be holding a channel_tcp directly instead of the socket + tcp_endpoint.   Then we can eliminate duplicate data like the node_id, tcp_endpoint, node shared_ptr etc.

This can be a separate PR though.

Reply:
Replaced partially. remote_endpoint remains to remove response_channel on closing until we are 100% sure that socket returning correct endpoint if remote peer closed connection. Also removing node shared pointer & replacing with channel node reference cause some random segfaults as it is